[pcl] pcl[surface-on-nurbs] build failure

### Operating system

Windows11

### Compiler

MSVC vc143

### Steps to reproduce the behavior

`vcpkg install  pcl[apps,cuda,openni2,qt,simulation,surface-on-nurbs,tools,visualization,vtk]`

### Failure logs

Installing 1/1 pcl[apps,core,cuda,opengl,openni2,qt,simulation,surface-on-nurbs,tools,visualization,vtk]:x64-windows@1.14.1#1...
Building pcl[apps,core,cuda,opengl,openni2,qt,simulation,surface-on-nurbs,tools,visualization,vtk]:x64-windows@1.14.1#1...
-- Using cached PointCloudLibrary-pcl-pcl-1.14.1.tar.gz.
-- Extracting source C:/Users/user/opt/vcpkg/downloads/PointCloudLibrary-pcl-pcl-1.14.1.tar.gz
-- Applying patch add-gcc-version-check.patch
-- Applying patch fix-check-sse.patch
-- Applying patch fix-numeric-literals-flag.patch
-- Applying patch install-layout.patch
-- Applying patch install-examples.patch
-- Applying patch fix-clang-cl.patch
-- Applying patch gh-5985-inline.patch
-- Applying patch io_ply.patch
-- Using source at C:/Users/user/opt/vcpkg/buildtrees/pcl/src/pcl-1.14.1-9b54503ead.clean
-- Found external ninja('1.11.0').
-- Configuring x64-windows
-- Building x64-windows-dbg
-- Building x64-windows-rel
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: C:/Users/user/scoop/apps/cmake/3.30.5/bin/cmake.exe --build . --config Release --target install -- -v -j13
    Working Directory: C:/Users/user/opt/vcpkg/buildtrees/pcl/x64-windows-rel
    See logs for more information:
      C:\Users\user\opt\vcpkg\buildtrees\pcl\install-x64-windows-rel-out.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_install.cmake:16 (vcpkg_cmake_build)
  ports/pcl/portfile.cmake:77 (vcpkg_cmake_install)
  scripts/ports.cmake:192 (include)

### Additional context

[install-x64-windows-rel-out.log](https://github.com/user-attachments/files/17393271/install-x64-windows-rel-out.log)

Not a port bug.
~~~
C:\Users\user\opt\vcpkg\buildtrees\pcl\src\pcl-1.14.1-9b54503ead.clean\surface\src\3rdparty\opennurbs\opennurbs_lookup.cpp(675) : fatal error C1001: Internal compiler error.
(compiler file 'D:\a\_work\1\s\src\vctools\Compiler\Utc\src\p2\main.c', line 247)
 To work around this problem, try simplifying or changing the program near the locations listed above.
If possible please provide a repro here: https://developercommunity.visualstudio.com 
Please choose the Technical Support command on the Visual C++ 
 Help menu, or open the Technical Support help file for more information
  cl!RaiseException()+0x6c
  cl!RaiseException()+0x6c
  cl!InvokeCompilerPass()+0x25f
  cl!InvokeCompilerPassW()+0x14607b
  cl!CloseTypeServerPDB()+0x209c9
  cl!CloseTypeServerPDB()+0x24924
  cl!CloseTypeServerPDB()+0x2386d
  cl!InvokeCompilerPassW()+0x6b80b
  cl!recalloc()+0xa3
  cl!BaseThreadInitThunk()+0x1d
  cl!RtlUserThreadStart()+0x28
~~~

--- a/ports/pcl/0012-msvc-optimizer-workaround.patch
+++ b/ports/pcl/0012-msvc-optimizer-workaround.patch
@@ -1,0 +1,28 @@
+diff --git a/surface/src/3rdparty/opennurbs/opennurbs_lookup.cpp b/surface/src/3rdparty/opennurbs/opennurbs_lookup.cpp
+index 13584fbbdb3..e006d3a5667 100644
+--- a/surface/src/3rdparty/opennurbs/opennurbs_lookup.cpp
++++ b/surface/src/3rdparty/opennurbs/opennurbs_lookup.cpp
+@@ -666,9 +666,13 @@ std::size_t ON_SerialNumberMap::ActiveIdCount() const
+   return m_active_id_count;
+ }
+ 
+-#if (_MSC_VER >= 1930 && _MSC_VER <= 1939)
++#if (_MSC_VER >= 1930 && _MSC_VER <= 1949)
+ // Solves internal compiler error on MSVC 2022
+ // (see https://github.com/microsoft/vcpkg/issues/19561)
++#define ON_VS2022_COMPILER_CRASH
++#endif
++
++#if defined(ON_VS2022_COMPILER_CRASH)
+ #pragma optimize("", off)
+ #endif
+ struct ON_SerialNumberMap::SN_ELEMENT* ON_SerialNumberMap::FirstElement() const
+@@ -722,7 +726,7 @@ struct ON_SerialNumberMap::SN_ELEMENT* ON_SerialNumberMap::FirstElement() const
+   }
+   return e;
+ }
+-#if (_MSC_VER >= 1930 && _MSC_VER <= 1939)
++#if defined(ON_VS2022_COMPILER_CRASH)
+ #pragma optimize("", on)
+ #endif
+ 

--- a/ports/pcl/portfile.cmake
+++ b/ports/pcl/portfile.cmake
@@ -15,6 +15,7 @@ vcpkg_from_github(
         io_ply.patch
         6053.diff # https://github.com/PointCloudLibrary/pcl/pull/6053
         6990a3b0d7dd3c1ca04a1a473cc172a937418060.diff # https://github.com/PointCloudLibrary/pcl/pull/6105
+        0012-msvc-optimizer-workaround.patch # backport pcl #6143 and #6154
 )
 
 string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "dynamic" PCL_SHARED_LIBS)

--- a/ports/pcl/vcpkg.json
+++ b/ports/pcl/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "pcl",
   "version": "1.14.1",
-  "port-version": 2,
+  "port-version": 3,
   "description": "Point Cloud Library (PCL) is open source library for 2D/3D image and point cloud processing.",
   "homepage": "https://github.com/PointCloudLibrary/pcl",
   "license": "BSD-3-Clause",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -6870,7 +6870,7 @@
     },
     "pcl": {
       "baseline": "1.14.1",
-      "port-version": 2
+      "port-version": 3
     },
     "pcre": {
       "baseline": "8.45",

--- a/versions/p-/pcl.json
+++ b/versions/p-/pcl.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "57795e21c0c37576db6bfb1df3b134de461d10e0",
+      "version": "1.14.1",
+      "port-version": 3
+    },
+    {
       "git-tree": "00b339d9b6b61b9f0742cfea4c482e70e9ac5ad8",
       "version": "1.14.1",
       "port-version": 2

